test/v1/models/request/submitBsas/NonFurnishedHolidayLetSpec.scala

```diff
@@ -0,0 +1,165 @@
+      |         "residentialFinancialCost": 1000.49,
+      |         "other": 1000.49
+      |
+      |      }
+      |}
+      |""".stripMargin)
+
+  val inputWithExpensesOnlyJson: JsValue = Json.parse(
+    """
+      |{
+      |      "expenses": {
+      |         "premisesRunningCosts": -1000.49,
+      |         "repairsAndMaintenance": 1000.49,
+      |         "financialCosts": 1000.49,
+      |         "professionalFees": 1000.49,
+      |         "travelCosts": 1000.49,
+      |         "costOfServices": -1000.49,
+      |         "residentialFinancialCost": 1000.49,
+      |         "other": 1000.49
+      |      }
+      |}
+      |""".stripMargin)
+
+  val requestJson: JsValue = Json.parse(
+    """
+      |{
+      |      "income": {
+      |         "totalRentsReceived": "1000.49",
+      |         "premiumsOfLeaseGrant": "1000.49",
+      |         "reversePremiums": "1000.49",
+      |         "otherPropertyIncome": "1000.49"
+      |      },
+      |      "expenses": {
+      |         "premisesRunningCosts": "-1000.49",
+      |         "repairsAndMaintenance": "1000.49",
+      |         "financialCosts": "1000.49",
+      |         "professionalFees": "1000.49",
+      |         "travelCosts": "1000.49",
+      |         "costOfServices": "-1000.49",
+      |         "residentialFinancialCost": "1000.49",
+      |         "other": "1000.49"
+      |      }
+      |}
+      |""".stripMargin)
+
+  val invalidJson: JsValue = Json.parse(
+    """
+      |{
+      |      "income": {
+      |         "rentIncome": "1000.49",
+      |         "premiumsOfLeaseGrant": "1000.49",
+      |         "reversePremiums": 1000.49,
+      |         "otherPropertyIncome": "1000.49"
+      |      },
+      |      "expenses": {
+      |         "premisesRunningCosts": true,
+      |         "repairsAndMaintenance": "1000.49",
+      |         "financialCosts": "1000.49",
+      |         "professionalFees": "1000.49",
+      |         "travelCosts": "1000.49",
+      |         "costOfServices": -1000.49,
+      |         "residentialFinancialCost": "1000.49",
+      |         "other": "1000.49"
+      |      }
+      |}
+      |""".stripMargin)
+
+  val model: NonFurnishedHolidayLet = NonFurnishedHolidayLet(
+    Some(NonFHLIncome(
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49)
+    )),
+    Some(NonFHLExpenses(
+      Some(-1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(-1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49)
+    ))
+  )
+
+  val modelWithExpensesOnly: NonFurnishedHolidayLet = NonFurnishedHolidayLet(
+    None,
+    Some(NonFHLExpenses(
+      Some(-1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(-1000.49),
+      Some(1000.49),
+      Some(1000.49),
+      Some(1000.49)
+    ))
+  )
+
+  "NonFurnishedHolidayLet" when {
```

Comment:
The success tests are all failing because the JSON doesn't contain `consolidatedExpenses` yet you're looking for it in the model.